### ISSUE TITLE
Convert depth map to point cloud

### ISSUE THREAD

**Issue by caobingquan6** [realsenseai/realsense-ros#2086]
Hello, the camera model I am using is L515. I want to know how the depth map and point cloud are converted in the code. But I did not find a detailed calculation process in the code.

I found that the conversion is directly obtained by the following code. But please answer for the specific conversion process.

<rs_frame.h>
/**
* When called on Points frame type, this method returns a pointer to an array of 3D vertices of the model
* The coordinate system is: X right, Y up, Z away from the camera. Units: Meters
* \param[in] frame       Points frame
* \param[out] error      If non-null, receives any error that occurs during this call, otherwise, errors are ignored
* \return                Pointer to an array of vertices, lifetime is managed by the frame
*/
rs2_vertex* rs2_get_frame_vertices(const rs2_frame* frame, rs2_error** error);

In other words, I want to understand the camera parameters and detailed calculation process of the depth map and point cloud conversion.

**Comment by MartyG-RealSense** [comment]
Hi @caobingquan6   A better way to calculate a point cloud that is generated using vertices would be to use **points.get_vertices**, a process that is described step by step in the SDK's C++ pointcloud tutorial guide linked to below.  Full source code for the project can also be found at the top of that page.

https://github.com/IntelRealSense/librealsense/tree/master/examples/pointcloud

**Comment by caobingquan6** [comment]
Yes, I saw this. But I want to know what is the specific calculation process of points.get_vertices()? Because the conversion of depth images into point clouds requires computation. But I did not find this calculation process.


What I found is as follows:

<rs_frame.hpp>
        const vertex* get_vertices() const
        {
            rs2_error* e = nullptr;
            auto res = rs2_get_frame_vertices(get(), &e);
            error::handle(e);
            return (const vertex*)res;
        }

Then:
<rs_frame.h>
/**
* When called on Points frame type, this method returns a pointer to an array of 3D vertices of the model
* The coordinate system is: X right, Y up, Z away from the camera. Units: Meters
* \param[in] frame       Points frame
* \param[out] error      If non-null, receives any error that occurs during this call, otherwise, errors are ignored
* \return                Pointer to an array of vertices, lifetime is managed by the frame
*/
rs2_vertex* rs2_get_frame_vertices(const rs2_frame* frame, rs2_error** error);


However, the above process does not have a detailed calculation process. This is what I need.

**Comment by MartyG-RealSense** [comment]
I'm not aware of a reference that describes points.get_vertices in the detail that you have in mind, though further details about how it works can be found in https://github.com/IntelRealSense/librealsense/issues/1783

**Comment by caobingquan6** [comment]
In other words, we can convert the depth image into a point cloud through the camera's internal parameter matrix (such as focal length, aperture center, etc.). Where can I get this information?

**Comment by MartyG-RealSense** [comment]
The L515 data sheet document provides the technical details of the L515's color sensor.

https://dev.intelrealsense.com/docs/lidar-camera-l515-datasheet

![image](https://user-images.githubusercontent.com/41145062/134389304-8e9648e4-2395-4df9-ba0a-a1ae51828e43.png)

You can also access the L515's intrinsic and extrinsic information in real-time with the rs-enumerate-devices tool's calibration mode by launching the tool with the instruction **rs-enumerate-devices -c**

![image](https://user-images.githubusercontent.com/41145062/134390534-8d0a83c3-fe83-4415-8d5e-c89f100d5c40.png)

**Comment by MartyG-RealSense** [comment]
Hi @caobingquan6  Do you require further assistance with this case, please?  Thanks!

**Comment by MartyG-RealSense** [comment]
Case closed due to no further comments received.
